### PR TITLE
docs: "user guide" section cleanup

### PR DESCRIPTION
**Pull request by zilto**
## Changes
- Renamed the section from "How To Guides" to "User Guide".
- Changed the title of "User Guide" pages to be easier to visually scan. Essentially, this consists of shorter titles with the most important word at the beginning
- changed the order of user guide items
- updated the list of integrations

Before:
image

After:
image

## How I tested this
- build docs locally

## Checklist

- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future TODOs are captured in comments
- [x] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: docs/how-tos/custom-driver.rst
Comment (skrawcz) [inline docs/how-tos/custom-driver.rst:1]: why the deletion?

File: docs/how-tos/async.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ==============================
-Run Hamilton in a Microservice
+Online transformations
```
Comment (skrawcz) [inline docs/how-tos/async.rst:2]: I think micro-service is better here than online transformation. The main point to take away is that you can run Hamilton in a web-service.

Comment (skrawcz) [inline docs/how-tos/async.rst:2]:
```suggestion
Micro-service
```

File: docs/how-tos/custom-driver.rst
Comment (skrawcz) [reply docs/how-tos/custom-driver.rst:1]: This could be renamed to "Custom Driver"

File: docs/how-tos/load-data.rst [hunk 2]
```diff
@@ -1,12 +0,0 @@
-==================
```
Comment (skrawcz) [inline docs/how-tos/load-data.rst:1]: even though this section isn't great, I think a `loading data` guide makes sense that points to examples/concepts (and even the recording from today).

File: docs/how-tos/use-without-pandas.rst [hunk 3]
```diff
@@ -1,14 +0,0 @@
-===========================
```
Comment (skrawcz) [inline docs/how-tos/use-without-pandas.rst:1]: fine to delete.

**Comment by skrawcz** [comment]
To me the purpose of the `how-to` section is for goal oriented people - "I want to do X with Hamilton"  where X is a broad concept -- for integrations X is a specific narrower technology/topic. 

So I think the changes mostly look good. Left comments where appropriate.

**Comment by zilto** [comment]
> To me the purpose of the `how-to` section is for goal oriented people - "I want to do X with Hamilton" where X is a broad concept -- while the integrations section is when X is a specific narrower technology/topic.

I agree on the purpose of the "how-to" / "user guide" section, but I think a lot of is too introductory / generic. In that case, it should reframed to be part of the "Concepts" section. The few good pages include "feature engineering", "pre-commit", "jupyter notebooks".

For instance, I believe "User guide" pages should mirror frequent Q/A on Slack

**Review by skrawcz — APPROVED** [review]

**Pull request merged**